docs/reference/extensions/gnome-extension.rst

```diff
@@ -0,0 +1,212 @@
+                target: $SNAP/gpu-2404
+                default-provider: mesa-2404
+
+    .. group-tab:: core22
+
+        .. code-block:: yaml
+
+            plugs:
+              desktop:
+                  mount-host-font-cache: false
+              gtk-3-themes:
+                  interface: content
+                  target: $SNAP/data-dir/themes
+                  default-provider: gtk-common-themes
+              icon-themes:
+                  interface: content
+                  target: $SNAP/data-dir/icons
+                  default-provider: gtk-common-themes
+              sound-themes:
+                  interface: content
+                  target: $SNAP/data-dir/sounds
+                  default-provider: gtk-common-themes
+              gnome-42-2204:
+                  interface: content
+                  target: $SNAP/gnome-platform
+                  default-provider: gnome-42-2204
+
+.. code-block:: yaml
+
+    plugs:
+      - desktop
+      - desktop-legacy
+      - gsettings
+      - opengl
+      - wayland
+      - x11
+      - mount-observe
+      - calendar-service
+
+
+Included packages
+-----------------
+
+The GNOME extension is derived from two separate snaps -- a `build snap
+<https://github.com/ubuntu/gnome-sdk/blob/gnome-42-2204-sdk/snapcraft.yaml>`_ and a
+`platform snap
+<https://github.com/ubuntu/gnome-sdk/blob/gnome-42-2204/snapcraft.yaml>`_.
+
+The build snap compiles libraries from source that are commonly used across GNOME apps.
+Examples include GLib, GTK, and gnome-desktop. These are built to provide newer versions
+of these packages that exist in either the core24 or core22 base snaps (a subset of
+their respective Ubuntu archives).
+
+The platform snap takes the build snap and makes all of those libraries available at
+build time to snaps using this extension. This way, snap authors don't need to include
+the pieces of the build snap that are unnecessary at runtime, like compilers, in the
+final snap.
+
+
+Included environment variables
+------------------------------
+
+In addition to using the build and platform snaps, this extension sets several
+environment variables, links, and default plugs for the app to use, and a default
+build-environment for each part in your snap to use.
+
+
+Build variables
+~~~~~~~~~~~~~~~
+
+The following build enviornment variables are included in each part built in a snap that
+this this extension.
+
+You can declare additional variables in the ``build-environment`` key. Furthermore,
+these default variables can be overridden by declaring them in the recipe.
+
+The paths slightly differ between core24 and core22 bases.
+
+.. tabs::
+
+    .. group-tab:: core24
+
+        .. code-block:: yaml
+
+            build-environment:
+              - PATH: /snap/gnome-46-2404-sdk/current/usr/bin${PATH:+:$PATH}
+              - XDG_DATA_DIRS: $CRAFT_STAGE/usr/share:/snap/gnome-46-2404-sdk/current/usr/share:/usr/share${XDG_DATA_DIRS:+:$XDG_DATA_DIRS}
```

Comment:
@mr-cal I wasn't sure about this particular format for showing included env vars. I understand that it could be useful to frame it as YAML, but it rubs me the wrong way and I feel like in a pinch someone could confuse the meaning here.

Do you think it would be better to switch this to bash syntax?

Reply:
We should format these as normal code.